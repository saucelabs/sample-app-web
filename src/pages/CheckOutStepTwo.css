.checkout_summary_container {
  max-width: 1280px;
  margin: 0 auto;
  padding: 0 15px 30px;
}

.summary_info_label {
  border-top: 2px solid #efefef;
  padding: 20px 0 0;
}

.summary_value_label {
  font-weight: 800;
  padding: 5px 0 20px 0;
}

.summary_subtotal_label {
  border-top: 2px solid #efefef;
  padding: 20px 0 10px;
}

.summary_tax_label {
  padding: 0;
}

.summary_total_label {
  padding: 10px 0;
  font-weight: 800;
  margin: 0 0 8px 0;
}

.summary_info_label,
.summary_value_label,
.summary_subtotal_label,
.summary_tax_label,
.summary_total_label {
  font-size: 16px;
  margin-left: 50px;
}

@media only screen and (min-width: 900px) {
  .summary_info_label,
  .summary_value_label,
  .summary_subtotal_label,
  .summary_tax_label,
  .summary_total_label {
    margin-left: 80px;
  }

  .summary_info_label {
    padding: 40px 0 0;
  }

  .summary_value_label {
    padding: 5px 0 40px 0;
  }

  .summary_subtotal_label {
    padding: 40px 0 10px;
  }
}
