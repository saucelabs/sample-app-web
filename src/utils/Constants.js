export const FONT_FAMILY = "Roboto, Arial, Helvetica, sans-serif";
export const VALID_USERNAMES = [
  "standard_user",
  "locked_out_user",
  "problem_user",
  "performance_glitch_user",
  "error_user",
  "visual_user",
];
export const VALID_PASSWORD = "secret_sauce";
export const ROUTES = {
  LOGIN: "/",
  INVENTORY: "/inventory.html",
  INVENTORY_LIST: "/inventory-item.html",
  CART: "/cart.html",
  CHECKOUT_STEP_ONE: "/checkout-step-one.html",
  CHECKOUT_STEP_TWO: "/checkout-step-two.html",
  CHECKOUT_COMPLETE: "/checkout-complete.html",
};
export const CART_CONTENTS = "cart-contents";
export const SESSION_USERNAME = "session-username";
