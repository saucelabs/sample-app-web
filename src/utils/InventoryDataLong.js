export const InventoryDataLong = [
  {
    id: 0,
    name: "Sauce Labs Bike Light",
    desc: `A red light isn't the desired state in testing but it sure helps when riding your bike at night. Water-resistant with 3 lighting modes, 1 AAA battery included.`,
    price: 9.99,
    image_url: "bike-light-1200x1500.jpg",
  },
  {
    id: 1,
    name: "Sauce Labs Bolt T-Shirt (XS)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
  {
    id: 2,
    name: "Sauce Labs Onesie",
    desc: `Rib snap infant onesie for the junior automation engineer in development. Reinforced 3-snap bottom closure, two-needle hemmed sleeved and bottom won't unravel.`,
    price: 7.99,
    image_url: "red-onesie-1200x1500.jpg",
  },
  {
    id: 3,
    name: "Test.allTheThings() T-Shirt (Red) (XS)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },
  {
    id: 4,
    name: "Sauce Labs Backpack",
    desc: `carry.allTheThings() with the sleek, streamlined Sly Pack that melds uncompromising style with unequaled laptop and tablet protection.`,
    price: 29.99,
    image_url: "sauce-backpack-1200x1500.jpg",
  },
  {
    id: 5,
    name: "Sauce Labs Fleece Jacket (XS)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 6,
    name: "Sauce Labs Fleece Jacket (S)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 7,
    name: "Sauce Labs Fleece Jacket (M)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 8,
    name: "Sauce Labs Fleece Jacket (L)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 9,
    name: "Sauce Labs Fleece Jacket (XL)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 10,
    name: "Sauce Labs Fleece Jacket (XXL)",
    desc: `It's not every day that you come across a midweight quarter-zip fleece jacket capable of handling everything from a relaxing day outdoors to a busy day at the office.`,
    price: 49.99,
    image_url: "sauce-pullover-1200x1500.jpg",
  },
  {
    id: 11,
    name: "Test.allTheThings() T-Shirt (Red) (S)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },

  {
    id: 12,
    name: "Test.allTheThings() T-Shirt (Red) (M)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },

  {
    id: 13,
    name: "Test.allTheThings() T-Shirt (Red) (L)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },

  {
    id: 14,
    name: "Test.allTheThings() T-Shirt (Red) (L)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },

  {
    id: 15,
    name: "Test.allTheThings() T-Shirt (Red) (XL)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },

  {
    id: 16,
    name: "Test.allTheThings() T-Shirt (Red) (XXL)",
    desc: `This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.`,
    price: 15.99,
    image_url: "red-tatt-1200x1500.jpg",
  },
  {
    id: 17,
    name: "Sauce Labs Bolt T-Shirt (M)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
  {
    id: 18,
    name: "Sauce Labs Bolt T-Shirt (L)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
  {
    id: 19,
    name: "Sauce Labs Bolt T-Shirt (XL)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
  {
    id: 20,
    name: "Sauce Labs Bolt T-Shirt (XXL)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
  {
    id: 21,
    name: "Sauce Labs Bolt T-Shirt (S)",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
    image_url: "bolt-shirt-1200x1500.jpg",
  },
];
