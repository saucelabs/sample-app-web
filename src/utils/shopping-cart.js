export class ShoppingCart {
  static addItem(itemId) {
    // pull out our current cart contents
    const curContents = ShoppingCart.getCartContents();

    /* istanbul ignore else */
    if (curContents.indexOf(itemId) < 0) {
      // Item's not yet present - add it now
      curContents.push(itemId);

      // We modified our cart, so store it now
      ShoppingCart.setCartContents(curContents);
    }
  }

  static removeItem(itemId) {
    // pull out our current cart contents
    const curContents = ShoppingCart.getCartContents();
    const itemIndex = curContents.indexOf(itemId);

    /* istanbul ignore else */
    if (itemIndex >= 0) {
      // Remove this item from the array
      curContents.splice(itemIndex, 1);

      // We modified our cart, so store it now
      ShoppingCart.setCartContents(curContents);
    }
  }

  static isItemInCart(itemId) {
    // pull out our current cart contents
    const curContents = ShoppingCart.getCartContents();

    // If the item is in the array, return true
    return curContents.indexOf(itemId) >= 0;
  }

  static getCartContents() {
    // pull out our current cart contents
    let curContents = window.localStorage.getItem("cart-contents");

    // Make an empty list if this is the first item
    if (curContents == null) {
      curContents = [];
    } else {
      // We have an existing cart, so deserialize it now since localStorage stores in JSON strings
      curContents = JSON.parse(curContents);
    }

    return curContents;
  }

  static setCartContents(newContents) {
    window.localStorage.setItem("cart-contents", JSON.stringify(newContents));

    // Notify our listeners
    /* istanbul ignore next */
    ShoppingCart.LISTENERS.forEach((curListener) => {
      curListener.forceUpdate();
    });
  }

  static resetCart() {
    window.localStorage.removeItem("cart-contents");

    // Notify our listeners
    /* istanbul ignore next */
    ShoppingCart.LISTENERS.forEach((curListener) => {
      curListener.forceUpdate();
    });
  }

  static registerCartListener(handler) {
    ShoppingCart.LISTENERS.push(handler);
  }
}

ShoppingCart.LISTENERS = [];
