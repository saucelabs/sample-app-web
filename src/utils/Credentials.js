import Cookies from "js-cookie";
import { SESSION_USERNAME, VALID_PASSWORD, VALID_USERNAMES } from "./Constants";

/**
 * Verify the credentials
 *
 * @param {string} username
 * @param {string} password
 *
 * @return {boolean}
 */
export function verifyCredentials(username, password) {
  if (password !== VALID_PASSWORD) {
    return false;
  }

  return VALID_USERNAMES.includes(username);
}

/**
 * Store the data in our cookies
 *
 * @param {string} username
 *
 * @param {string} password
 */
export function setCredentials(username, password) {
  let date = new Date();
  date.setTime(date.getTime() + 10 * 60 * 1000);

  Cookies.set(SESSION_USERNAME, username, { expires: date });
}

/**
 * Remove the credentials
 */
export function removeCredentials() {
  Cookies.remove(SESSION_USERNAME);
}

/**
 * Return current logged username
 *
 * @return {string | undefined}
 */
export function currentUser() {
  return Cookies.get(SESSION_USERNAME);
}

/**
 * Check if this is a problem user
 *
 * @return {boolean}
 */
export function isProblemUser() {
  return Cookies.get(SESSION_USERNAME) === "problem_user";
}

/**
 * Check if this is a performance user
 *
 * @return {boolean}
 */
export function isPerformanceGlitchUser() {
  return Cookies.get(SESSION_USERNAME) === "performance_glitch_user";
}

/**
 * Check if this a logged out user
 *
 * @return {boolean}
 */
export function isLockedOutUser() {
  return Cookies.get(SESSION_USERNAME) === "locked_out_user";
}

/**
 * Check if this is an error user
 *
 * @return {boolean}
 */
export function isErrorUser() {
  return Cookies.get(SESSION_USERNAME) === "error_user";
}

/**
 * Check if the user is logged in with a valid username
 *
 * @return {boolean}
 */
export function isLoggedIn() {
  const sessionUsername = Cookies.get(SESSION_USERNAME);
  const isValidUsername = VALID_USERNAMES.includes(sessionUsername);

  return isValidUsername && sessionUsername !== "locked_out_user";
}

/**
 * Check if this is a visual user
 *
 * @return {boolean}
 */
export function isVisualUser() {
  return Cookies.get(SESSION_USERNAME) === "visual_user";
}
