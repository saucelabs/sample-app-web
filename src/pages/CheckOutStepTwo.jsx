import React from "react";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";
import { isProblemUser } from "../utils/Credentials";
import { ROUTES } from "../utils/Constants";
import { ShoppingCart } from "../utils/shopping-cart";
import { InventoryData } from "../utils/InventoryData";
import CartItem from "../components/CartItem";
import SwagLabsFooter from "../components/Footer";
import HeaderContainer from "../components/HeaderContainer";
import Button, { BUTTON_SIZES, BUTTON_TYPES } from "../components/Button";
import "./CheckOutStepTwo.css";

const CheckOutStepTwo = ({ history }) => {
  const clearCart = () => {
    /* istanbul ignore else */
    // No cart clear on order complete for the problem user
    if (!isProblemUser()) {
      // Wipe out our shopping cart
      ShoppingCart.resetCart();
    }
  };
  const contents = ShoppingCart.getCartContents();
  let orderTotal = 0;

  for (const curItem in contents) {
    orderTotal = orderTotal + InventoryData[contents[curItem]].price;
    if (isProblemUser()) {
      // double up for the problem user
      orderTotal = orderTotal + InventoryData[contents[curItem]].price;
    }
  }

  const orderTax = (orderTotal * 0.08).toFixed(2);

  return (
    <div id="page_wrapper" className="page_wrapper">
      <div id="contents_wrapper">
        <HeaderContainer secondaryTitle="Checkout: Overview" />
        <div
          id="checkout_summary_container"
          className="checkout_summary_container"
        >
          <div>
            <div className="cart_list">
              <div className="cart_quantity_label">QTY</div>
              <div className="cart_desc_label">DESCRIPTION</div>
              {contents.map((item, i) => {
                return <CartItem key={i} item={InventoryData[item]} />;
              })}
            </div>
            <div className="summary_info">
              <div className="summary_info_label">Payment Information:</div>
              <div className="summary_value_label">SauceCard #31337</div>
              <div className="summary_info_label">Shipping Information:</div>
              <div className="summary_value_label">
                FREE PONY EXPRESS DELIVERY!
              </div>
              <div className="summary_subtotal_label">
                Item total: ${orderTotal}
              </div>
              <div className="summary_tax_label">Tax: ${orderTax}</div>
              <div className="summary_total_label">
                Total: ${(orderTotal + parseFloat(orderTax)).toFixed(2)}
              </div>
              <div className="cart_footer">
                <Button
                  // `cart_cancel_link` has no style function
                  // but is there for backwards compatibility
                  customClass="cart_cancel_link"
                  label="Cancel"
                  onClick={(evt) => {
                    evt.preventDefault();
                    history.push(ROUTES.INVENTORY);
                  }}
                  size={BUTTON_SIZES.MEDIUM}
                  testId="cancel"
                  type={BUTTON_TYPES.BACK}
                />
                <Button
                  customClass="cart_button"
                  label="Finish"
                  onClick={(evt) => {
                    evt.preventDefault();
                    clearCart();
                    history.push(ROUTES.CHECKOUT_COMPLETE);
                  }}
                  size={BUTTON_SIZES.MEDIUM}
                  testId="finish"
                  type={BUTTON_TYPES.ACTION}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <SwagLabsFooter />
    </div>
  );
};
CheckOutStepTwo.propTypes = {
  /**
   * The history
   */
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default withRouter(CheckOutStepTwo);
