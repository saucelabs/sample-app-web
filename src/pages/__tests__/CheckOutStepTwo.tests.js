import React from "react";
import { shallow } from "enzyme";
import CheckOutStepTwo from "../CheckOutStepTwo";
import { ShoppingCart } from "../../utils/shopping-cart";
import * as Credentials from "../../utils/Credentials";

jest.mock("../../utils/shopping-cart");

let props;

describe("CheckOutStepTwo", () => {
  beforeEach(() => {
    props = {
      history: { push: jest.fn() },
    };
    ShoppingCart.getCartContents = jest.fn().mockReturnValue([]);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly without any items", () => {
    const wrapper = shallow(<CheckOutStepTwo.WrappedComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly items", () => {
    const cartContents = [1, 2, 3];
    ShoppingCart.getCartContents = jest.fn().mockReturnValue(cartContents);
    const wrapper = shallow(<CheckOutStepTwo.WrappedComponent {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should redirect when trying to cancel", () => {
    const wrapper = shallow(<CheckOutStepTwo.WrappedComponent {...props} />);
    const ContinueShopping = wrapper.find("Button").at(0);
    ContinueShopping.simulate("click", {
      preventDefault() {},
    });

    expect(props.history.push).toBeCalledWith("/inventory.html");
  });

  it("should redirect when trying to finish", () => {
    const wrapper = shallow(<CheckOutStepTwo.WrappedComponent {...props} />);
    const Checkout = wrapper.find("Button").at(1);
    Checkout.simulate("click", {
      preventDefault() {},
    });

    expect(ShoppingCart.resetCart).toHaveBeenCalledTimes(1);
    expect(props.history.push).toBeCalledWith("/checkout-complete.html");
  });

  it("should give the incorrect order total when we are logged in as a problem user", () => {
    const isProblemUserSpy = jest.spyOn(Credentials, "isProblemUser");
    isProblemUserSpy.mockReturnValue(true);
    const cartContents = [1, 2, 3];
    ShoppingCart.getCartContents = jest.fn().mockReturnValue(cartContents);
    const wrapper = shallow(<CheckOutStepTwo.WrappedComponent {...props} />);

    expect(isProblemUserSpy).toHaveBeenCalledTimes(3);
    expect(wrapper).toMatchSnapshot();
  });

  it("should should not reset the cart when a problem user finishes", () => {
    const isProblemUserSpy = jest.spyOn(Credentials, "isProblemUser");
    isProblemUserSpy.mockReturnValue(true);

    const wrapper = shallow(<CheckOutStepTwo.WrappedComponent {...props} />);
    const Checkout = wrapper.find("Button").at(1);
    Checkout.simulate("click", {
      preventDefault() {},
    });

    expect(isProblemUserSpy).toHaveBeenCalledTimes(1);
    expect(ShoppingCart.resetCart).not.toHaveBeenCalled();
    expect(props.history.push).toBeCalledWith("/checkout-complete.html");
  });
});
