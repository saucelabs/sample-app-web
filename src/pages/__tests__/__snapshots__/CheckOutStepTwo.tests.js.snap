// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckOutStepTwo should give the incorrect order total when we are logged in as a problem user 1`] = `
<div
  className="page_wrapper"
  id="page_wrapper"
>
  <div
    id="contents_wrapper"
  >
    <HeaderContainer
      secondaryTitle="Checkout: Overview"
    />
    <div
      className="checkout_summary_container"
      id="checkout_summary_container"
    >
      <div>
        <div
          className="cart_list"
        >
          <div
            className="cart_quantity_label"
          >
            QTY
          </div>
          <div
            className="cart_desc_label"
          >
            DESCRIPTION
          </div>
          <withRouter(CartItem)
            item={
              Object {
                "desc": "Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.",
                "id": 1,
                "image_url": "bolt-shirt-1200x1500.jpg",
                "name": "Sauce Labs Bolt T-Shirt",
                "price": 15.99,
              }
            }
            key="0"
          />
          <withRouter(CartItem)
            item={
              Object {
                "desc": "Rib snap infant onesie for the junior automation engineer in development. Reinforced 3-snap bottom closure, two-needle hemmed sleeved and bottom won't unravel.",
                "id": 2,
                "image_url": "red-onesie-1200x1500.jpg",
                "name": "Sauce Labs Onesie",
                "price": 7.99,
              }
            }
            key="1"
          />
          <withRouter(CartItem)
            item={
              Object {
                "desc": "This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.",
                "id": 3,
                "image_url": "red-tatt-1200x1500.jpg",
                "name": "Test.allTheThings() T-Shirt (Red)",
                "price": 15.99,
              }
            }
            key="2"
          />
        </div>
        <div
          className="summary_info"
        >
          <div
            className="summary_info_label"
          >
            Payment Information:
          </div>
          <div
            className="summary_value_label"
          >
            SauceCard #31337
          </div>
          <div
            className="summary_info_label"
          >
            Shipping Information:
          </div>
          <div
            className="summary_value_label"
          >
            FREE PONY EXPRESS DELIVERY!
          </div>
          <div
            className="summary_subtotal_label"
          >
            Item total: $
            79.94
          </div>
          <div
            className="summary_tax_label"
          >
            Tax: $
            6.40
          </div>
          <div
            className="summary_total_label"
          >
            Total: $
            86.34
          </div>
          <div
            className="cart_footer"
          >
            <Button
              customClass="cart_cancel_link"
              label="Cancel"
              onClick={[Function]}
              size="medium"
              testId="cancel"
              type="secondary back"
            />
            <Button
              customClass="cart_button"
              label="Finish"
              onClick={[Function]}
              size="medium"
              testId="finish"
              type="action"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <SwagLabsFooter />
</div>
`;

exports[`CheckOutStepTwo should render correctly items 1`] = `
<div
  className="page_wrapper"
  id="page_wrapper"
>
  <div
    id="contents_wrapper"
  >
    <HeaderContainer
      secondaryTitle="Checkout: Overview"
    />
    <div
      className="checkout_summary_container"
      id="checkout_summary_container"
    >
      <div>
        <div
          className="cart_list"
        >
          <div
            className="cart_quantity_label"
          >
            QTY
          </div>
          <div
            className="cart_desc_label"
          >
            DESCRIPTION
          </div>
          <withRouter(CartItem)
            item={
              Object {
                "desc": "Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.",
                "id": 1,
                "image_url": "bolt-shirt-1200x1500.jpg",
                "name": "Sauce Labs Bolt T-Shirt",
                "price": 15.99,
              }
            }
            key="0"
          />
          <withRouter(CartItem)
            item={
              Object {
                "desc": "Rib snap infant onesie for the junior automation engineer in development. Reinforced 3-snap bottom closure, two-needle hemmed sleeved and bottom won't unravel.",
                "id": 2,
                "image_url": "red-onesie-1200x1500.jpg",
                "name": "Sauce Labs Onesie",
                "price": 7.99,
              }
            }
            key="1"
          />
          <withRouter(CartItem)
            item={
              Object {
                "desc": "This classic Sauce Labs t-shirt is perfect to wear when cozying up to your keyboard to automate a few tests. Super-soft and comfy ringspun combed cotton.",
                "id": 3,
                "image_url": "red-tatt-1200x1500.jpg",
                "name": "Test.allTheThings() T-Shirt (Red)",
                "price": 15.99,
              }
            }
            key="2"
          />
        </div>
        <div
          className="summary_info"
        >
          <div
            className="summary_info_label"
          >
            Payment Information:
          </div>
          <div
            className="summary_value_label"
          >
            SauceCard #31337
          </div>
          <div
            className="summary_info_label"
          >
            Shipping Information:
          </div>
          <div
            className="summary_value_label"
          >
            FREE PONY EXPRESS DELIVERY!
          </div>
          <div
            className="summary_subtotal_label"
          >
            Item total: $
            39.97
          </div>
          <div
            className="summary_tax_label"
          >
            Tax: $
            3.20
          </div>
          <div
            className="summary_total_label"
          >
            Total: $
            43.17
          </div>
          <div
            className="cart_footer"
          >
            <Button
              customClass="cart_cancel_link"
              label="Cancel"
              onClick={[Function]}
              size="medium"
              testId="cancel"
              type="secondary back"
            />
            <Button
              customClass="cart_button"
              label="Finish"
              onClick={[Function]}
              size="medium"
              testId="finish"
              type="action"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <SwagLabsFooter />
</div>
`;

exports[`CheckOutStepTwo should render correctly without any items 1`] = `
<div
  className="page_wrapper"
  id="page_wrapper"
>
  <div
    id="contents_wrapper"
  >
    <HeaderContainer
      secondaryTitle="Checkout: Overview"
    />
    <div
      className="checkout_summary_container"
      id="checkout_summary_container"
    >
      <div>
        <div
          className="cart_list"
        >
          <div
            className="cart_quantity_label"
          >
            QTY
          </div>
          <div
            className="cart_desc_label"
          >
            DESCRIPTION
          </div>
        </div>
        <div
          className="summary_info"
        >
          <div
            className="summary_info_label"
          >
            Payment Information:
          </div>
          <div
            className="summary_value_label"
          >
            SauceCard #31337
          </div>
          <div
            className="summary_info_label"
          >
            Shipping Information:
          </div>
          <div
            className="summary_value_label"
          >
            FREE PONY EXPRESS DELIVERY!
          </div>
          <div
            className="summary_subtotal_label"
          >
            Item total: $
            0
          </div>
          <div
            className="summary_tax_label"
          >
            Tax: $
            0.00
          </div>
          <div
            className="summary_total_label"
          >
            Total: $
            0.00
          </div>
          <div
            className="cart_footer"
          >
            <Button
              customClass="cart_cancel_link"
              label="Cancel"
              onClick={[Function]}
              size="medium"
              testId="cancel"
              type="secondary back"
            />
            <Button
              customClass="cart_button"
              label="Finish"
              onClick={[Function]}
              size="medium"
              testId="finish"
              type="action"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <SwagLabsFooter />
</div>
`;
